GoogleChromeLabs/ps-analysis-tool
# Embed Web Stories in PSAT

## Description
This PR aims to add an explorable explanations section where user can toggle and see explanations for topics like cookies, protected audience etc in Web Stories format.

## Relevant Technical Choices
- Since manifest V3, chrome doesn't allow injection of scripts that are remotely hosted. To achieve this, all the required scripts by the stories have been bundled with the extension and the URLs have been replaced when the story markup is rendered.

## Testing Instructions
- Clone this branch.
- In the terminal run `npm i && npm run build:ext`
- Now go to example.com and open the DevTools and open Privacy Sandbox tab.
- You should see a section in sidebar titled as `Explorable Explanation` with Web-Stories Icon.
- Click on the section. A set of stories with their hero image should be rendered on the main content.
- Clicking on the hero image will result in the story opening in a light box.

## Additional Information:



## Screenshot/Screencast

https://github.com/user-attachments/assets/7911db78-e933-402c-90ee-753ef49de6f7




---

## Checklist
